### PR TITLE
fix(perf): avoid unnecessary strlen

### PR DESCRIPTION
**Pull request by cyyever**

### CODE REVIEW

**Review by Skylion007 — APPROVED** [review]
Neat! These are on some pretty hot code paths so these changes should be helpful.

File: include/pybind11/attr.h [hunk 1]
```diff
@@ -377,7 +377,7 @@ template <> struct process_attribute<is_new_style_constructor> : process_attribu
 };
 
 inline void process_kw_only_arg(const arg &a, function_record *r) {
-    if (!a.name || strlen(a.name) == 0)
+    if (!a.name || a.name[0] == '\0')
```
Comment (henryiii) [inline include/pybind11/attr.h:380]: Would this be cleaner this way?
```suggestion
    if (!a.name || !a.name[0])
```

Maybe the first way reads better, though. Just a thought.

Comment (Skylion007) [reply include/pybind11/attr.h:380]: I am personally against implicit boolean conversions for readability reasons.

Comment (henryiii) [reply include/pybind11/attr.h:380]: Yeah, I don't think I like it that way either. The current way clearly indicates what it's doing.

**Review by henryiii — APPROVED** [review]

File: include/pybind11/attr.h (hunk 1 above)
Comment (cyyever) [reply include/pybind11/attr.h:380]: Both ways are equivalent. However, I also want to make the intention more explicit.

**Review by rwgk — APPROVED** [review]
We fixed the Windows issue a few days ago. Could you please rebase on current master?  We'll merge this when we see that the CI is green. (I'm tempted to just merge but better to not be too wild west.)

**Comment by cyyever** [comment]
> We fixed the Windows issue a few days ago. Could you please rebase on current master? We'll merge this when we see that the CI is green. (I'm tempted to just merge but better to not be too wild west.)
OK

**Comment by rwgk** [comment]
I think the one macos straggler will just time out again. Happens all the time recently. Safe to ignore.

**Pull request merged**